Gordon360/Services/RecIM/TeamService.cs

```diff
@@ -0,0 +1,176 @@
+                            Win = st.Win,
+                            // for now Loss is calculated with query, just for no dummy data added to database
+                            Loss = t.MatchTeam
+                                .Where(mt => mt.Match.StatusID == 6
+                                && mt.Score < _context.MatchTeam
+                                .FirstOrDefault(opmt => opmt.MatchID == mt.MatchID
+                                && opmt.TeamID != teamID)
+                                .Score)
+                                .Count(),
+                            Tie = t.MatchTeam
+                                .Where(mt => mt.Match.StatusID == 6
+                                && mt.Score == _context.MatchTeam
+                                .FirstOrDefault(opmt => opmt.MatchID == mt.MatchID
+                                && opmt.TeamID != teamID)
+                                .Score)
+                                .Count(),
+                        }),
+                }).FirstOrDefault();
+            return team;
+        }
+
+        public async Task<int> PostTeamAsync(TeamUploadViewModel t, string username)
+        {
+            var participantID = _context.ACCOUNT
+                                .FirstOrDefault(a => a.AD_Username == username)
+                                .account_id;
+
+            var team = new Team
+            {
+                Name = t.Name,
+                StatusID = t.StatusID,
+                ActivityID = t.ActivityID,
+                Private = t.Private,
+                Logo = t.Logo,
+            };
+            await _context.Team.AddAsync(team);
+            await _context.SaveChangesAsync();
+
+            await AddUserToTeamAsync(team.ID, username);
+            await UpdateParticipantRoleAsync(team.ID, participantID, 5);
+
+            return team.ID;
+        }
+
+        public async Task UpdateParticipantRoleAsync(int teamID, int participantID, int participantRoleID)
+        {
+            var participantTeam = _context.ParticipantTeam.FirstOrDefault(pt => pt.ParticipantID == participantID && pt.TeamID == teamID);
+            participantTeam.RoleType = participantRoleID;
+
+            await _context.SaveChangesAsync();
+        }
+
+        public async Task<int> UpdateTeamAsync(int teamID, TeamPatchViewModel update)
+        {
+            var t = await _context.Team.FindAsync(teamID);
+            t.Name = update.Name ?? t.Name;
+            t.StatusID = update.StatusID ?? t.StatusID;
+            t.Private = (bool)update.Private;
+            t.Logo = update.Logo ?? t.Logo;
+
+            await _context.SaveChangesAsync();
+
+            return teamID;
+        }
+        public async Task AddUserToTeamAsync(int teamID, string username)
+        {
+            var participantID = _context.ACCOUNT
+                                .FirstOrDefault(a => a.AD_Username == username)
+                                .account_id;
```

Comment:
Make sure to handle the case where the username is invalid. You would want to stop and return some error.

Reply:
In fact, I would say that when you want to get an account by username, you should be using `AccountService.GetAccountByUsername` instead of using `_context.Account` directly.

Reply:
I wondered about that @EjPlatzer.

Reply:
> In fact, I would say that when you want to get an account by username, you should be using `AccountService.GetAccountByUsername` instead of using `_context.Account` directly.

I actually did not think to really use our AccountService for my participant routes so I'll make that adjustment in the next PR